Centaurus.Domain/WebSockets/Alpha/Info/InfoWebSocketConnection.cs

```diff
@@ -0,0 +1,131 @@
+                        await webSocket.CloseAsync(status, desc, CancellationToken.None);
+                    }
+                    catch (WebSocketException exc)
+                    {
+                        //ignore client disconnection
+                        if (exc.WebSocketErrorCode != WebSocketError.ConnectionClosedPrematurely)
+                            throw;
+                    }
+                }
+                Dispose();
+            }
+            catch (Exception e)
+            {
+                logger.Error(e);
+            }
+            finally
+            {
+                OnClosed?.Invoke(this, new EventArgs { });
+            }
+        }
+
+        public event EventHandler OnClosed;
+
+        public async Task Listen()
+        {
+            try
+            {
+                var reader = await webSocket.GetString();
+                while (true)
+                {
+                    BaseCommand command = null;
+                    try
+                    {
+                        command = BaseCommand.Deserialize(reader);
+                        var result = await commandHandlers.HandleCommand(this, command);
+                        await SendMessage(result);
+                    }
+                    catch (Exception exc)
+                    {
+                        if (exc is ConnectionCloseException)
+                            throw;
+                        var statusCode = exc.GetStatusCode();
+                        await SendMessage(new ErrorResponse
+                        {
+                            RequestId = command?.RequestId ?? 0,
+                            Status = (int)statusCode,
+                            Error = (int)statusCode < 500 ? exc.Message : statusCode.ToString()
+                        });
+
+                        if (statusCode == ResultStatusCodes.InternalError || !Global.IsAlpha)
+                            logger.Error(exc);
+                    }
+                    reader = await webSocket.GetString();
+                }
+            }
+            catch (ConnectionCloseException e)
+            {
+                await CloseConnection(e.Status, e.Description);
+            }
+            catch (WebSocketException e)
+            {
+                var closureStatus = WebSocketCloseStatus.InternalServerError;
+                if (e.WebSocketErrorCode == WebSocketError.ConnectionClosedPrematurely)
+                    closureStatus = WebSocketCloseStatus.NormalClosure;
+                else
+                    logger.Error(e);
+                await CloseConnection(closureStatus);
+            }
+            catch (Exception e)
+            {
+                logger.Error(e);
+            }
+        }
+
+        public async Task SendMessage(object message)
+        {
+            await webSocket.SendAsync(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(message)), WebSocketMessageType.Binary, true, CancellationToken.None);
```

Comment:
Are you sure about `WebSocketMessageType.Binary`? JSON is a text format, so you should probably send and receive it as text, not the binary datagrams. Better for debugging and feels right in general.